### PR TITLE
[2/x] Move veNft data to its own project type and change when it is fetched

### LINKED ISSUE #1529: Move veNft data to its own context and change how it is fetched
## Description

Currently veNft data is loaded in with the main project context. However, since its data is not currently used anywhere but the veNft drawer it would likely make sense to give it its own context and fetch when a user opens the drawer.

## Proposal

Move veNft data to its own context and change when it is retrieved.

### PR DESCRIPTION
**Pull request by jmilldotdev**
## What does this PR do and why?

Removes veNft context from V2ProjectContext and puts it into its own context type. It is fetched when the user opens the drawer and not at the project level.

Closes #1529 

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by jmilldotdev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1586** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1586" target="_blank">Graphite</a> 
    * **PR #1587** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1587" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1587?utm_source=stack-comment).

**Review by tomquirk — APPROVED** [review]

**Comment by jmilldotdev** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1587?utm_source=stack-comment).


Job Reference: 5peKCVtRG7hIeC6BSrBI

**Comment by jmilldotdev** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1587?utm_source=stack-comment).


Job Reference: 0agyc4HkFZeecGjH4TTU

**Pull request merged**
